tutorials/content_filtering/c++11/home_automation_publisher.cxx

```diff
@@ -0,0 +1,39 @@
+//
+// (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+//
+//  RTI grants Licensee a license to use, modify, compile, and create derivative
+//  works of the Software.  Licensee has the right to distribute object form
+//  only for use with RTI products.  The Software is provided "as is", with no
+//  warranty of any type, including any warranty for fitness for any purpose.
+//  RTI is under no obligation to maintain or support the Software.  RTI shall
+//  not be liable for any incidental or consequential damages arising out of the
+//  use or inability to use the software.
+//
+
+#include <iostream>
+#include <thread>
+#include <rti/rti.hpp>
+#include "home_automation.hpp"
+
+void publish_sensor(
+    const std::string& sensor_name,
+    const std::string& room_name)
+{
+    dds::domain::DomainParticipant participant(0);
+    dds::topic::Topic<DeviceStatus> topic(participant, "WindowStatus");
+    dds::pub::DataWriter<DeviceStatus> writer(topic);
+
+    DeviceStatus device_status { sensor_name, room_name, false };
+    for (int i = 0; i < 1000; i++) {
+        device_status.is_open(!device_status.is_open());
+        writer.write(device_status);
```

Comment:
Shouldn't we print something so the user can see that samples are being written?

Reply:
We can do that in the GitHub examples, but I want to keep the snippets in the guide free of clutter. (It's ok to keep them slightly different for things like these.)

Reply:
I will do the the change in other PR since this is already approved and it is not something that only affects this module.